tests/integration-service/integration.go

```diff
@@ -380,5 +383,81 @@ var _ = framework.IntegrationServiceSuiteDescribe("Integration Service E2E tests
 
 			})
 		})
+
+		Describe("valid dtcls doesn't exist", Ordered, func() {
+			BeforeAll(func() {
+				// Initialize the tests controllers
+				f, err = framework.NewFramework(IntegrationServiceUser)
+				Expect(err).NotTo(HaveOccurred())
+
+				createApp()
+				createComponent()
+
+				integrationTestScenario, err = f.AsKubeAdmin.IntegrationController.CreateIntegrationTestScenarioWithEnvironment(applicationName, appStudioE2EApplicationsNamespace, BundleURL, InPipelineName, EnvironmentName)
+				Expect(err).ShouldNot(HaveOccurred())
+				env, err = f.AsKubeAdmin.IntegrationController.CreateEnvironment(appStudioE2EApplicationsNamespace, EnvironmentName)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			AfterAll(func() {
+				if !CurrentSpecReport().Failed() {
+					cleanup()
+
+					Expect(f.AsKubeAdmin.GitOpsController.DeleteAllEnvironmentsInASpecificNamespace(appStudioE2EApplicationsNamespace, 30*time.Second)).To(Succeed())
+					Expect(f.AsKubeAdmin.IntegrationController.DeleteSnapshot(snapshot_push, appStudioE2EApplicationsNamespace)).To(BeNil())
+				}
+			})
+
+			It("valid deploymentTargetClass doesn't exist", func() {
+				validDTCLS, err := f.AsKubeAdmin.IntegrationController.HaveAvailableDeploymentTargetClassExist()
+				Expect(validDTCLS).To(BeNil())
+				Expect(err).To(BeNil())
+			})
+
+			It("creates a snapshot of push event", func() {
+				sampleImage := "quay.io/redhat-appstudio/sample-image@sha256:841328df1b9f8c4087adbdcfec6cc99ac8308805dea83f6d415d6fb8d40227c1"
+				snapshot_push, err = f.AsKubeAdmin.IntegrationController.CreateSnapshot(applicationName, appStudioE2EApplicationsNamespace, componentName, sampleImage)
+				Expect(err).ShouldNot(HaveOccurred())
+				GinkgoWriter.Printf("snapshot %s is found\n", snapshot_push.Name)
+			})
+
+			When("nonexisting valid deploymentTargetClass", func() {
+				It("check no GitOpsCR is created for the dtc with nonexisting deploymentTargetClass", func() {
+					timeout = time.Second * 600
+					interval = time.Second * 10
+					Eventually(func() (bool, error) {
+						spaceRequestList, err := f.AsKubeAdmin.IntegrationController.GetSpaceRequests(appStudioE2EApplicationsNamespace)
+						if err != nil || len(spaceRequestList.Items) > 0 {
+							return false, err
+						}
```

Comment:
why do we need to do polling here? if we do find a spaceRequest then this "if" condition will be true and we'll go over this `Eventually` block again and again until either we exceed the `interval` or suddenly the spaceRequest gets deleted from the cluster.

Reply:
@dheerajodha Good point, thanks. We don't need to poll them again and again. And are there tests running in parallel when running them?

Reply:
> And are there tests running in parallel when running them?

I'm not sure about that, but based on the [conversation that happened here](https://redhat-internal.slack.com/archives/C02FANRBZQD/p1685040162953479) recently, I think tests do run in parallel.

Reply:
Edit: Yes, CI runs them in parallel, please see [this line in the latest prow link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_e2e-tests/527/pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e/1661252755534123008#1:build-log.txt%3A1040).

Reply:
@dheerajodha Thanks a lot! Updated.